patterns/ecs-v1/rails

```diff
@@ -1,13 +1,13 @@
 RUUID \h{32}
 # rails controller with action
-RCONTROLLER (?<controller>[^#]+)#(?<action>\w+)
+RCONTROLLER (?<[rails][controller][class]>[^#]+)#(?<[rails][controller][action]>\w+)
 
 # this will often be the only line:
-RAILS3HEAD (?m)Started %{WORD:verb} "%{URIPATHPARAM:request}" for %{IPORHOST:clientip} at (?<timestamp>%{YEAR}-%{MONTHNUM}-%{MONTHDAY} %{HOUR}:%{MINUTE}:%{SECOND} %{ISO8601_TIMEZONE})
+RAILS3HEAD (?m)Started %{WORD:[http][request][method]} "%{URIPATHPARAM:[url][original]}" for %{IPORHOST:[client][address]} at (?<timestamp>%{YEAR}-%{MONTHNUM}-%{MONTHDAY} %{HOUR}:%{MINUTE}:%{SECOND} %{ISO8601_TIMEZONE})
 # for some a strange reason, params are stripped of {} - not sure that's a good idea.
-RPROCESSING \W*Processing by %{RCONTROLLER} as (?<format>\S+)(?:\W*Parameters: {%{DATA:params}}\W*)?
-RAILS3FOOT Completed %{NUMBER:response}%{DATA} in %{NUMBER:totalms}ms %{RAILS3PROFILE}%{GREEDYDATA}
-RAILS3PROFILE (?:\(Views: %{NUMBER:viewms}ms \| ActiveRecord: %{NUMBER:activerecordms}ms|\(ActiveRecord: %{NUMBER:activerecordms}ms)?
+RPROCESSING \W*Processing by %{RCONTROLLER} as (?<[rails][request][format]>\S+)(?:\W*Parameters: {%{DATA:[rails][request][params]}}\W*)?
+RAILS3FOOT Completed %{POSINT:[http][response][status_code]:int}%{DATA} in %{NUMBER:[rails][request][duration][total]:float}ms %{RAILS3PROFILE}%{GREEDYDATA}
```

Comment:
I would map `[rails][request][duration][total]` to `[event][duration]`. Note that you'd have to convert to nanos.

Reply:
yeah I did see that use-case in beats (doing the nanos conversion).
however for LS there's no added support "above" the patterns (no `logstash-input-rails_log`), like in beats.
users will have to handle the full ECS "transition" on their own, if they use these grok patterns (except for syslog we have an input there that consumes these).

so I was hesitant to use `event.duration`, think it's better if we do not populate since it expects nanos ...

Reply:
Yeah agreed. You shouldn't use event.duration if there's no oppt to convert to nanos